components/style/values/specified/length.rs

```diff
@@ -458,6 +458,17 @@ impl Parse for Length {
     }
 }
 
+impl Either<Length, None_> {
+    #[inline]
+    #[allow(missing_docs)]
+    pub fn parse_non_negative_length(context: &ParserContext, input: &mut Parser) -> Result<Self, ()> {
+        if input.try(|input| None_::parse(context, input)).is_ok() {
+            return Ok(Either::Second(None_));
+        }
+        Length::parse_internal(input, AllowedNumericType::NonNegative).map(Either::First)
```

Comment:
Why can't we change existing `parse_non_negative_length` impls to something like this?

``` rust
impl<T: Parse> Either<Length, T> {
    pub fn parse_non_negative_length(..) { }
}
```

They all do the same thing anyway.